### ISSUE TITLE
ArgumentError: Content-Length not given and Transfer-Encoding is not `chunked'

### ISSUE THREAD

**Issue by simi** [Jesus/dropbox_api#26]
Hello, I wasn't able to upload any file getting this exception.

```
ArgumentError: Content-Length not given and Transfer-Encoding is not `chunked'
from /home/retro/.rubies/ruby-2.2.6/lib/ruby/2.2.0/net/http/generic_request.rb:192:in `send_request_with_body_stream'
```

After a little lookup, I was able to fix this by changing `build_request` in `ContentUpload`.

When I passed one of `'Content-Length' => content.size` or `'Transfer-Encoding' => 'chunked'` as an additional header, everything worked as expected and file was uploaded without problems (verified by downloading and md5 hashing).

Does it make sense to include one of these headers by default or at least add new option for `upload` method?

I can submit PR once this will be decided.

*Thanks for this awesome gem!*

**Comment by Jesus** [comment]
Hi simi,

I can't reproduce the problem you're describing. What version of Faraday are you using?

**Comment by simi** [comment]
I'm using Faraday `0.12.1`.

**Comment by simi** [comment]
fixed in #30
